frame/staking/src/lib.rs

```diff
@@ -1655,17 +1837,27 @@ impl<T: Trait> Module<T> {
 	}
 }
 
+/// In this implementation `new_session(session)` must be called before `end_session(session-1)`
+/// i.e. the new session must be planned before the ending of the previous session.
+///
+/// `new_session(0)` must be called at genesis.
```

Comment:
I think we should support less constraint, allow staking module to be plugged in a runtime that already have started. And also that the start of the era is not necessarily the first block when staking module has been put in the runtime.

Reply:
Currently the staking module is required to be run from genesis. I think this is a constraint we don't want. I'm working on this.